docs/docs-beta/docs/guides/data-modeling/partitioning.md

```diff
@@ -87,14 +87,38 @@ In this example:
 
 Now that you've seen how to model partitioned assets in different ways, this section shows how to define dependencies between various partitioned assets, and between partitioned assets and un-partitioned assets.
 
-### Dependencies between time-based partitions
-
 Partitioned assets in Dagster can have dependencies on other partitioned assets. This allows you to create complex data pipelines where the output of one partitioned asset feeds into another. Here's how it works:
 
 - A downstream asset can depend on one or more partitions of an upstream asset
 - The partitioning schemes don't need to be identical, but they should be compatible
 
-{/* TODO */}
+### Dependencies between different time-based partitions
+
+In [Define time-partitioned assets](#define-time-partitioned-assets), we created two time-partitioned assets: `daily_sales_data` and `daily_sales_summary`, which can be executed at the same time in a single schedule.
+
+However, sometimes you might want to define dependencies between different time-based partitions. For example, you might want to aggregate daily data into a weekly report.
+
+In this case, we have a `weekly_sales_summary` asset that depends on the `daily_sales_data` asset. Here's how to set up dependencies between different time-based partitions:
+
+<CodeExample filePath="guides/data-modeling/partitioning/time_based_partition_dependencies.py" language="python" title="Time-based partition dependencies" />
+
+In this example:
+
+- We have a `daily_sales_data` asset partitioned by day, which will be executed daily.
+- The `weekly_sales_summary` asset depends on the `daily_sales_data` asset, which will be executed weekly.
+
+  - In this asset, the weekly partition depends on all its parent partitions (all seven days of the week). We use `context.asset_partition_key_range_for_input("daily_sales_data")` to get a range of partition keys, which includes the start and end of the week.
+
+- To automate the execution of these assets:
+
+  - First, we specify `automation_condition=AutomationCondition.eager()` to the `weekly_sales_summary` asset. This ensures it runs weekly after all seven daily partitions of `daily_sales_data` are up-to-date.
+  - Second, we specify `automation_condition=AutomationCondition.cron(cron_schedule="0 1 * * *")` to the `daily_sales_data` asset. This ensures it runs daily.
+
+Note: In [a simpler example](#define-time-partitioned-assets) above, we manually set up a daily schedule for asset execution. For more complex dependency logic, it's recommended to use automation conditions instead of schedules. Automation conditions specify when an asset should run, which allows you to define execution criteria without custom scheduling logic. For more details, see [Declarative Automation](/concepts/automation/declarative-automation).
```

Comment:
I'm a little torn on this approach, as it seems like its "sneaking" a new concept into this guide. Maybe we should leave this section using schedules for the daily and weekly asset, and then in a different guide focus on "alternative to schedules: automation conditions"

Reply:
Aren't Automation Conditions experimental? We should not be using experimental APIs in our guides/documentation until they're stable. Not to say we shouldn't have a guide on Automation Condition, but we should introduce experimental APIs into other places.

Reply:
Thank you for fulfilling the purpose of this PR! While I liked how this piece turned out, I was wary that this may introduce DA too earlier (both in terms of timeline and content structure). Happy to take DA out and go back to the traditional two schedule ways.

@slopp the limitation of one daily schedule and another weekly schedule is that the weekly schedule would just "hope" all seven daily ones have succeeded at the time it's started. I guess it's totally fine in the guide, but should we link to DA in a note to explain how DA can address this limitation?

Reply:
This comes up fairly frequently and I wish it was an easier thing to handle (even in a DA-less world)

I think a minimal option would be to address via docs as you suggest 

Another option would be to show a schedule decorated function that first checks for the upstream partitions ... 
this gets dangerously close to multi-asset sensor territory though which is a PITA and worse than DAs to explain

Reply:
> This comes up fairly frequently and I wish it was an easier thing to handle (even in a DA-less world)

I wonder if the right answer going forward is to go DA in this case. My optimistic understanding is once DA is stable, this is exactly one of what DA is designed for, so we may not need to have a DA-less solution going forward. With that assumption, the minimal option should work while we're in the transition period.

> Another option would be to show a schedule decorated function that first checks for the upstream partitions ...

I actually started with writing this before learning DA, but felt too much to explain. Maybe could be a good use case example in the use-case-repo? (that place feels lower stake than guides which feel much more like "the official recommendations")

Reply:
I have thought about this and am now okay with DA being here